src/formatDate.js

```diff
@@ -8,7 +8,56 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+
+  const dateParts = date.split(separatorFrom);
+
+  let year, month, day;
+
+  fromFormat.forEach((part, index) => {
+    if (part === 'YYYY' || part === 'YY') {
+      year = dateParts[index];
+    } else if (part === 'MM') {
+      month = dateParts[index];
+    } else if (part === 'DD') {
+      day = dateParts[index];
+    }
+  });
+
+  const convertYear = (inputYear, targetFormat) => {
+    if (targetFormat === 'YYYY') {
+      if (inputYear.length === 2) {
+        return Number(inputYear) < 30 ? '20' + inputYear : '19' + inputYear;
+      }
+
+      return inputYear;
+    } else if (targetFormat === 'YY') {
+      return inputYear.length === 4 ? inputYear.slice(-2) : inputYear;
+    }
+  };
+
+  if (toFormat.includes('YYYY') && year.length === 2) {
+    year = convertYear(year, 'YYYY');
+  } else if (toFormat.includes('YY') && year.length === 4) {
+    year = convertYear(year, 'YY');
```

Comment:
The conditional checks for year conversion seem redundant because you already handle the year conversion logic within the `convertYear` function. You could streamline the code by directly calling `convertYear` with the appropriate `targetFormat` based on the `toFormat` array without the need for these conditions.